src/ml_gym/frontend/dashboard/src/features/modelsStatus/modelsStatusSlice.tsx

```diff
@@ -0,0 +1,48 @@
+import { createSlice } from '@reduxjs/toolkit'
+import { ModelStatusMessageCollectionType, ModelStatusType, ModelStatusRowType } from '../../app/datatypes'
+import type { RootState } from '../../app/store';
+
+const initialState = {
+    messages: [],
+    experiment_id_to_latest_message_index: {}
+} as ModelStatusMessageCollectionType
+
+
+
+
+const modelsStatusSlice = createSlice({
+    name: 'modelsStatus',
+    initialState,
+    reducers: {
+        modelStatusAdded(state, action) {
+            const experiment_id: number = action.payload.data.payload.experiment_id
+            if (experiment_id in state.experiment_id_to_latest_message_index) {
+                const row_id = state.experiment_id_to_latest_message_index[experiment_id]
+                // prevents possible race condition
+                if (state.messages[row_id].event_id < action.payload.event_id) {
+                    state.messages[row_id] = action.payload
```

Comment:
Codacy has a fix for the issue: [Replace `····················state.messages[row_id]·=·action.payload` with `··········state.messages[row_id]·=·action.payload;`](https://app.codacy.com/gh/mlgym/mlgym/pullRequest?prid=9966992)
```suggestion
          state.messages[row_id] = action.payload;
```